docs/references/contribution_guide.md

```diff
@@ -0,0 +1,165 @@
+# Contribution Guide
+
+Welcome to **SGLang**! We appreciate your interest in contributing. This guide provides a concise overview of how to set up your environment, run tests, build documentation, and open a Pull Request (PR). Whether you’re fixing a small bug or developing a major feature, we encourage following these steps for a smooth contribution process.
+
+---
+
+## 1. Setting Up & Building from Source
+
+### 1.1 Fork and Clone the Repository
+
+> **Note**: SGLang does **not** accept PRs on the main repo. Please fork the repository under your GitHub account, then clone your fork locally.
+
+```bash
+git clone https://github.com/<your_user_name>/sglang.git
+cd sglang
+```
+
+### 1.2 Install Dependencies & Build
+
+Refer to our [Install SGLang](https://sgl-project.github.io/start/install.html) documentation for more details on setting up the necessary dependencies.
+
+Before installing packages like FlashInfer, ensure your local PyTorch and CUDA versions match:
+
+```bash
+python -c "import torch; print('PyTorch Version:', torch.__version__); print('CUDA Version:', torch.version.cuda)"
+nvidia-smi
+```
+
+Based on this information, choose the correct FlashInfer package (or any other CUDA-dependent library).
+Below is an example assuming you have **PyTorch 2.4** with **CUDA 12.4**:
+
+```bash
+# Example: Installing FlashInfer (Pick the right version for your CUDA/PyTorch)
+pip install --upgrade pip
+pip install -e "python[all]" --find-links https://flashinfer.ai/whl/cu124/torch2.4/flashinfer/
+
+# Now install the latest SGLang
+cd sglang/python
+pip install .
+```
+---
+
+## 2. Code Formatting with Pre-Commit
+
+We use [pre-commit](https://pre-commit.com/) to maintain consistent code style checks. Before pushing your changes, please run:
+
+```bash
+pip3 install pre-commit
+cd sglang
+pre-commit run --all-files
+```
+
+- **`pre-commit run --all-files`** manually runs all configured checks, applying fixes if possible.
+- If it fails the first time, re-run it to ensure lint errors are fully resolved.
+- Make sure your code passes all checks **before** creating a Pull Request.
+- **Do not commit** directly to the `main` branch. Always create a new branch (e.g., `feature/my-new-feature`), push your changes, and open a PR from that branch.
+
+---
+
+## 3. Writing Documentation & Running Docs CI
+
+Most documentation files are located under the `docs/` folder. We prefer **Jupyter Notebooks** over Markdown so that all examples can be executed and validated by our docs CI pipeline.
+
+### 3.1 Docs Workflow
+
```

Comment:
Add or update your Jupyter notebooks in the appropriate subdirectories under docs/.

If you add new files, remember to update index.rst (or relevant .rst files) accordingly.